cds_migrator_kit/videos/weblecture_migration/load/helpers.py

```diff
@@ -0,0 +1,211 @@
+        )
+        return object_version
+
+    except FileNotFoundError:
+        raise ManualImportRequired(f"File '{video_name}' not found.", stage="load")
+    except Exception as e:
+        raise ManualImportRequired(
+            f"Error uploading file to bucket'{video_name}': {e}", stage="load"
+        )
+
+
+def load_frames(payload, frame_paths):
+    """Load frames for the master video file without running the celery task.
+
+    payload (dict):
+        - version_id (str): The version ID of the main video file.
+        - key (str): The file name of the main video file.
+        - bucket_id (str): The bucket id of the record.
+        - deposit_id (str): The deposit id of the record.
+        - flow_id (str): The flow id.
+    frame_paths (list): A list of file paths of the frames.
+    """
+    # TODO sort frames_list?
+    try:
+        version_id = payload["version_id"]
+        object_version = as_object_version(version_id)
+
+        frames_task, kwargs = AVCFlowCeleryTasks.create_task(
+            ExtractFramesTask, payload, delete_copied=False
+        )
+        frames_task.flow_id = payload.get("flow_id")
+        # FramesTask Metadata
+        flow_task_metadata = frames_task.get_or_create_flow_task()
+        flow_task_metadata.status = FlowTaskStatus.STARTED
+
+        # Calculate time positions
+        options = frames_task._time_position(
+            duration=object_version.get_tags()["duration"],
+            frames_start=0,  # TODO they should be changed with using the frame file name?
+            frames_end=90,
+            frames_gap=10,
+        )
+        frames = frames_task._create_frames(
+            frame_paths,
+            object_version,
+            options.get("start_time"),
+            options.get("time_step"),
+        )
+        output_folder = tempfile.mkdtemp()
+        frames_task._create_gif(
+            bucket=str(object_version.bucket.id),
+            frames=frames,
+            output_dir=output_folder,
+            master_id=version_id,
+        )
+        flow_task_metadata.status = FlowTaskStatus.SUCCESS
+
+        # Cleanup
+        shutil.rmtree(output_folder)
+    except Exception:
+        shutil.rmtree(output_folder, ignore_errors=True)
+        frames_task.clean(version_id=version_id)
+        raise ManualImportRequired("Frame uploading failed!", stage="load")
+
+
+def load_subformats(payload, subformat_paths):
+    """Load subformats for the master video file without celery task.
+
+    payload (dict):
+        - version_id (str): The version ID of the main video file.
+        - key (str): The file name of the main video file.
+        - bucket_id (str): The bucket id of the record.
+        - deposit_id (str): The deposit id of the record.
+        - flow_id (str): The flow id.
+    subformat_paths (list): A list of file paths of the subformats.
+    """
+    # TODO should we need to check all the subformats exists? If missing should we generate?
```

Comment:
For the composite if I understood correctly it will be generated as 1080p, therefore ideally we should have 4 subformats (360, 480, 720 and 1080), I would add a check for this 4 and if not log it as you mentioned.